### PR TITLE
[M][REQ][JAVA]: Add option to generate arrays with "uniqueItems" true as Sets rather than Lists

### LINKED ISSUE #5254: [REQ][JAVA]: Add option to generate arrays with "uniqueItems" true as Sets rather than Lists
### Description
It would be great if there was an option to generate properties declared as "array" with "uniqueItems": true as a Set/HashSet rather than a List/ArrayList.

### Proposed solution
This could be enabled with a flag option to avoid it beying a breaking change.

### PR DESCRIPTION
**Pull request by Januson**
### Summary
 - Update Java code generation to use sets instead of lists when uniqueItems is set to true
 - Add import resolution for sets
 - Add tests

resolve #5254

### Java Technical Committee
 - @bbdouglas
 - @sreeshas
 - @jfiala
 - @lukoyanov
 - @cbornet
 - @jeff9finger
 - @karismann
 - @Zomzog
 - @lwlee2608
 - @bkabrda

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by bkabrda** [comment]
I'm wondering whether `HashSet` is the right Set implementation to use here. IIUC the marking a list with `uniqueItems: true` still means that it's a list, so ordering matters. I think `LinkedHashSet` would be a better implementation to use here.

**Comment by Januson** [comment]
@bkabrda Switched Set implementation to LinkedHashSet.

**Comment by Januson** [comment]
@bkabrda 
Fixed a couple issues where there was a missing import for Set/LinkedHashSet when uniqueItems was used for param/response.

Second issue was that some of the templates were expecting only list types, so I had to update those to support sets.

Now the generation of sets should work for models, parameters and responses.

File: samples/client/petstore/csharp-netcore/OpenAPIClient/src/Org.OpenAPITools/Api/FakeApi.cs [hunk 1]
```diff
@@ -257,7 +257,7 @@ public interface IFakeApiSync : IApiAccessor
         /// <param name="enumQueryString">Query parameter enum test (string) (optional, default to -efg)</param>
         /// <param name="enumQueryInteger">Query parameter enum test (double) (optional)</param>
         /// <param name="enumQueryDouble">Query parameter enum test (double) (optional)</param>
-        /// <param name="enumFormStringArray">Form parameter enum test (string array) (optional, default to $)</param>
```
Comment (wing328) [inline samples/client/petstore/csharp-netcore/OpenAPIClient/src/Org.OpenAPITools/Api/FakeApi.cs:260]: @Januson overall it's good but looks like it breaks the default value for array of enum (string). Can you please have a look?

Comment (Januson) [reply samples/client/petstore/csharp-netcore/OpenAPIClient/src/Org.OpenAPITools/Api/FakeApi.cs]: @wing328 Done.

**Comment by barnabycourt** [comment]
@wing328 Is there anything else that you would like from this fix? Thanks!

**Comment by wing328** [comment]
I've resolved the merge conflicts via https://github.com/OpenAPITools/openapi-generator/pull/5466/commits/08931f30b87ad75af6979744d3bee5b8dc5714c1

If all tests pass and no further feedback from anyone, I'll merge this PR.

**Pull request merged**